core/services/repo_manager.rb

```diff
@@ -107,17 +115,39 @@ def addRepo(file)
 
   private
 
-  # Find the latest available repository version
+  # Find all available product repositories by repository_key
   # @param product [Hash] hash array with information about product
   # @param repository_key [String] key of repository
-  def find_last_repository_version(product, repository_key)
-    all_available_repo = @repos.find_all do |elem|
+  def find_available_repo(product, repository_key)
+    @repos.find_all do |elem|
       elem[1]['product'] == product['name'] &&
         elem[1]['platform'] + '^' + elem[1]['platform_version'] == repository_key
     end
+  end
+
+  # Find the latest available repository version
+  # @param product [Hash] hash array with information about product
+  # @param repository_key [String] key of repository
+  def find_last_repository_version(product, repository_key)
+    all_available_repo = find_available_repo(product, repository_key)
     repo = all_available_repo.last[1]
     all_available_repo.delete_if { |elem| elem[1]['version'].include?('debug') }
     repo = all_available_repo.last[1] unless all_available_repo.nil?
     repo
   end
+
+  # Find the latest available repository version by major version if it's not exists.
+  # For example, for 10.2 version it returns latest 10.2.33-8 version.
+  # @param product [Hash] hash array with information about product
+  # @param repository_key [String] key of repository
+  def find_last_repository_by_major_version(product, repository_key)
+    return nil if product['version'].match(/^\d+\.\d+$/).nil?
```

Comment:
The product version may be 10.2.2 or 10. Your code should support both of them. Not only x.y version.